### PR TITLE
feat: update scripts, make it possible to run scripts of type *_from_…

### PR DESCRIPTION
**Pull request by vivanov00**
…list with proper parameters.

Will require improvements to documentation, to run the new scripts (v2.4.3) one must provide a lot more parameters as there has been significant drift.

### CODE REVIEW

File: bin/nexus_configurator [hunk 1]
```diff
@@ -93,8 +94,16 @@ def main():
         fatal("Failure to parse yaml file found at {}".format(args.config))
     for config_step in config:
         for script_type, resources in config_step.items():
-            for resource in resources:
-                resp = nx.run_script(script_type, **resource)
+            if "list" in script_type:
+                resp = nx.run_script_multi(script_type,resources)
```
Comment (stuart-warren) [inline bin/nexus_configurator]: nit: whitespace around comma

File: nexus_configurator/nexus.py [hunk 2]
```diff
@@ -111,6 +111,9 @@ def create_script(self, filename, name=None):
     def run_script(self, script, **kwargs):
         run_url = "script/{}/run".format(script)
         return self._post(run_url, kwargs, content_type="text/plain")
+    def run_script_multi(self, script,arg):
```
Comment (stuart-warren) [inline nexus_configurator/nexus.py]: nit: whitespace around comma

File: nexus_configurator/nexus.py [hunk 3]
```diff
@@ -111,6 +111,9 @@ def create_script(self, filename, name=None):
...
+      run_url = "script/{}/run".format(script)
+      return self._post(run_url, arg, content_type="text/plain")
```
Comment (stuart-warren) [inline nexus_configurator/nexus.py]: nit: indentation for these lines inconsistent with rest of file

**Review by stuart-warren — CHANGES_REQUESTED** [review]
few minor nits, otherwise looks good!

yet to validate

**Comment by vivanov00** [comment]
Nits - post-comma spacing and indentation for run_script_multi function body, fixed.

**Comment by vivanov00** [comment]
contrib/example_config.yaml updated.

File: contrib/example_config.yaml [hunk 4]
```diff
@@ -34,222 +34,246 @@
...
+  - name: maven-snapshot-group
+    type: group
+    format: maven2
+    blob_store: maven-snapshot-group
+    member_repos:
+      - maven-snapshots
+      - maven-central
+      - gradle-plugins
+      - jcenter
+      - maven-ocado-com
```
Comment (stuart-warren) [inline contrib/example_config.yaml]: can we remove these `maven-ocado-com` repos, see line 140 too

I don't see them created, so was probably my bad from previous commit

File: bin/nexus_configurator [hunk 5]
```diff
@@ -93,8 +94,16 @@ def main():
         fatal("Failure to parse yaml file found at {}".format(args.config))
     for config_step in config:
         for script_type, resources in config_step.items():
-            for resource in resources:
-                resp = nx.run_script(script_type, **resource)
+            if "list" in script_type:
```
Comment (stuart-warren) [inline bin/nexus_configurator]: I'd suggest this is: `if "_from_list" in script_type:`

File: bin/nexus_configurator [hunk 6]
```diff
@@ -8,8 +8,9 @@ import os.path
 from pkg_resources import resource_filename
 import sys
 import yaml
+import logging
 from nexus_configurator.nexus import Nexus, NexusUnauthorised, NexusConnectionError
-
+logging.basicConfig(level=logging.DEBUG)
```
Comment (stuart-warren) [inline bin/nexus_configurator]: so this is for requests?

I don't think debug really be the default

maybe something like this to set it dynamically https://github.com/ocadotechnology/cluster-location-webhook/blob/master/hook.py#L17-L22

**Review by stuart-warren — CHANGES_REQUESTED** [review]
`example_config.yaml` could probably be a lot shorter with fewer examples

No usage of `setup_users_from_list` which could be nice to use roles

everything works afaict

Otherwise looks good

File: contrib/example_config.yaml (hunk 4 above)
Comment (vivanov00) [reply contrib/example_config.yaml]: Yes, I shall remove them, and I agree on the shortness, but at the same time it is good to have lots of working examples.
Adding a setup_users_from_list section.

File: bin/nexus_configurator (hunk 6 above)
Comment (vivanov00) [reply bin/nexus_configurator]: Just something for local testing I added and forgot to take out as it wasn't really a detriment in my view.
Adding a '--debug' param would be nice to have  as well.

File: bin/nexus_configurator (hunk 5 above)
Comment (vivanov00) [reply bin/nexus_configurator]: Sure, there really isn't much of a danger of name collisions at the moment, but conciseness never hurts.

**Review by stuart-warren — APPROVED** [review]

**Pull request merged**